0.19.6 release post (#166)
TeamNewPipe/website: _posts/2020-07-14-newpipe-0.19.6-release.md

---
layout: post
title: "NewPipe 0.19.6 released: Audio/video desync bug and search improvements"
short: "NewPipe 0.19.6 released"
date: 2020-07-14 00:00:00 +0100
categories: [release, pinned]
author: gkeegan
image: newpipe
excerpt_separator: <!-- more -->
---

No more unsynced videos! This release will let your videos make sense and improve searching in NewPipe to make it more reliable and widespread.

<!-- more -->

## Additions

You can now add a playlist directly to your main page as a tab, a great change for those who frequently use NewPipe for music. <sup>[@Royosef in #3506](https://github.com/TeamNewPipe/NewPipe/pull/3506)</sup>

> There is an option to prevent subscriptions from being visible in the subscription picker for feeds which will be helpful to sort your subscriptions into different categories. <sup>[@mauriciocolli in #3404](https://github.com/TeamNewPipe/NewPipe/pull/3404)</sup>

Reviewer: Maybe clarify that subscriptions that are already in another feed won't be shown with this option enabled.

Author: That doesn't occur when they are in the default feed, correct?

Reviewer: Of course the default "All" feed is excluded, as that always contains all channels and can't be changed

Author: Ok good. Then this should do the trick:
```suggestion
There is an option to prevent subscriptions from being visible in the subscription picker for feeds which will be helpful to sort your subscriptions into different categories. If enabled, subscriptions that are already in another non-default feed will not be shown. <sup>[@mauriciocolli in #3404](https://github.com/TeamNewPipe/NewPipe/pull/3404)</sup>
```

Reviewer: That still isn't that clear imho, I was thinking more like the following: "There now is an option in the subscription picker to not show channels that are already in any feed (other than "All")."

Speaking (writing) of the subscription picker, it now has a search bar so you can easily find what you are looking for. <sup>[@mauriciocolli in #3373](https://github.com/TeamNewPipe/NewPipe/pull/3373)</sup>

> More search changes include "did you mean" & "showing result for" suggestions from YouTube so that you don't have to restart a search if you make a tpyo. <sup>[@Royosef in #3471](https://github.com/TeamNewPipe/NewPipe/pull/3471)</sup>

Reviewer: It's not YouTube-only and @B0pol made the appropriate changes on the NPE side

Author: ```suggestion
More search changes include "did you mean" & "showing result for" suggestions so that you don't have to restart a search if you make a tpyo. <sup>[@B0pol in Extractor #311](https://github.com/TeamNewPipe/NewPipeExtractor/pull/311) and [@Royosef in #3471](https://github.com/TeamNewPipe/NewPipe/pull/3471)</sup>
```

When using the background player, you can fast forward and rewind through the playback from within the menu for it (not through the notification). <sup>[@TheLastGimbus in #3473](https://github.com/TeamNewPipe/NewPipe/pull/3437)</sup>

> PeerTube videos are now able to be opened in NewPipe without explicitly sharing to it. The [privacytools.io](https://www.privacytools.io/) PeerTube instance was added to NewPipe. PeerTube embed links are also now able to be opened in NewPipe now. <sup>[@B0pol in #3406](https://github.com/TeamNewPipe/NewPipe/pull/3406), in [#3845](https://github.com/TeamNewPipe/NewPipe/pull/3845), and in [Extractor #344](https://github.com/TeamNewPipe/NewPipeExtractor/pull/344)</sup>

Reviewer: Like I said, no need to specifically mention PrivacyTools.io. Just say something like: "You will now be offered to open links from certain PeerTube instances in NewPipe when you open those in your browser or when you share those links."

Author: Is this better?

```suggestion
You will now be given a prompt to open links from certain PeerTube instances in NewPipe when they are opened in a browser or shared. PeerTube embed links are also now able to be opened in NewPipe now. <sup>[@B0pol in #3406](https://github.com/TeamNewPipe/NewPipe/pull/3406), in [#3845](https://github.com/TeamNewPipe/NewPipe/pull/3845), and in [Extractor #344](https://github.com/TeamNewPipe/NewPipeExtractor/pull/344)</sup>
```

Reviewer: Yes, though "are also now able to be opened" in the next sentence isn't really great, change that to "You could now also open …"

Author: That really is a lot of words to describe it. This is much shorter:
```suggestion
You will now be given a prompt to open links from certain PeerTube instances in NewPipe when they are opened in a browser or shared. Also, PeerTube embed links can be opened in NewPipe now. <sup>[@B0pol in #3406](https://github.com/TeamNewPipe/NewPipe/pull/3406), in [#3845](https://github.com/TeamNewPipe/NewPipe/pull/3845), and in [Extractor #344](https://github.com/TeamNewPipe/NewPipeExtractor/pull/344)</sup>
```

> More Invidious instances are added to the app manifest, so NewPipe will be suggested as an application if you open these links. <sup>[@B0pol in #3841](https://github.com/TeamNewPipe/NewPipe/pull/3841)</sup>

Reviewer: An average user doesn't know what the manifest is

Author: How about this?

```suggestion
More Invidious instances are supported for link opening so that NewPipe will be suggested as an application if you click those links. <sup>[@B0pol in #3841](https://github.com/TeamNewPipe/NewPipe/pull/3841)</sup>
```

Reviewer: Hmm… "for link opening" and "if you click those links" is kinda double

Author: Agreed. This seems better.
```suggestion
More Invidious instances are supported so that NewPipe will be suggested as an application if you open these links. <sup>[@B0pol in #3841](https://github.com/TeamNewPipe/NewPipe/pull/3841)</sup>
```

> There are even more translations added to NewPipe! These are: Arabic (Lybia), Bengali (India), Central Kurdish, and Javanese. As always, you can help with translations through [Weblate](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md#translation).

Reviewer: Why not link directly to Weblate?

Author: I didn't link directly to weblate since this provides some more info about it (such as how to add a new language.

Reviewer: OK, mark this as resolved then

## Bug fixes

> As mentioned early, the audio/video desync bug has been fixed. This bug was quite annoying to have to deal with when streaming video and now has been squashed. <sup>[@Stypox in #3837](https://github.com/TeamNewPipe/NewPipe/pull/3837)</sup>

Reviewer: *earlier

Author: ```suggestion
As mentioned earlier, the audio/video desync bug has been fixed. This bug was quite annoying to have to deal with when streaming video and now has been squashed. <sup>[@Stypox in #3837](https://github.com/TeamNewPipe/NewPipe/pull/3837)</sup>
```

More Invidious instances are added to the app manifest, so NewPipe will be suggested as an application if you open these links. <sup>[@B0pol in #3841](https://github.com/TeamNewPipe/NewPipe/pull/3841)</sup>

There are even more translations added to NewPipe! These are: Arabic (Lybia), Bengali (India), Central Kurdish, and Javanese. As always, you can help with translations through [Weblate](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md#translation).

## Bug fixes

As mentioned early, the audio/video desync bug has been fixed. This bug was quite annoying to have to deal with when streaming video and now has been squashed. <sup>[@Stypox in #3837](https://github.com/TeamNewPipe/NewPipe/pull/3837)</sup>

A crash that occurred when quickly swiping through search suggestions has been fixed. <sup>[@Stypox in #3759](https://github.com/TeamNewPipe/NewPipe/pull/3759)</sup>

> Channel names are now visible for all themes when in the background player. The text color used to be always black, which clearly did not work to well when using a dark theme. <sup>[@Stypox in #3822](https://github.com/TeamNewPipe/NewPipe/pull/3822)</sup>

Reviewer: Only in certain cases

Author: Sorry, what do you mean?

Reviewer: Only on certain phones the text color was black. In most cases it had the correct gray color.

Speaking of errors, there used to be a functionality that would remove an item from your playing queue if there is an error with its playback. This led to a temporary internet instability to cause many items to fail and thus being removed from the queue, making it possible for many items to be falsely removed. This functionality has been removed to fix the problem. <sup>[@Stypox in #3704](https://github.com/TeamNewPipe/NewPipe/pull/3704)</sup>

You're now able to long press on a video title to copy it to the clipboard. This is helpful when you are sharing a video somewhere where the link doesn't make an embed box to show other info about the video. <sup>[@adinilfeld in #3772](https://github.com/TeamNewPipe/NewPipe/pull/3772)</sup>

There is now more proper structural support for YouTube Mixes (so Mixes will be available in a future update, *not this one*). <sup>[@wb9688 in #3441](https://github.com/TeamNewPipe/NewPipe/pull/3441)</sup>

> Also for YouTube, NewPipe supports `start=` timestamps in links now so if you are sent a specific time in a long stream it will start playing there. <sup>[@wb9688 in Extractor #366](https://github.com/TeamNewPipe/NewPipeExtractor/pull/366)</sup>

Reviewer: Mention that `start=` is used by YT's embedded player

Author: ```suggestion
Also for YouTube, NewPipe supports `start=` timestamps (used by YouTube's embedded player) in links now so if you are sent a specific time in a long stream it will start playing there. <sup>[@wb9688 in Extractor #366](https://github.com/TeamNewPipe/NewPipeExtractor/pull/366)</sup>
```

The status bar now matches the toolbar color so NewPipe seems to fit in even more with your device. :) <sup>[@eames-palmer in #3774](https://github.com/TeamNewPipe/NewPipe/pull/3774)</sup>

NewPipe no longer writes metadata to downloaded files that designates they were downloaded via the NewPipe application. This saves a <small><small>tiny</small></small> amount of space and helps with privacy. <sup>[@kapodamy in #3843](https://github.com/TeamNewPipe/NewPipe/pull/3843)</sup>

> Debug APKs are now much smaller (by about 8MB) due to removing Checkstyle from them. This will make using a debug APK in order to help test new features much more friendly to those testing. <sup>[@wb9688 in #3828](https://github.com/TeamNewPipe/NewPipe/pull/3828)</sup>

Reviewer: *8 MB

Author: ```suggestion
Debug APKs are now much smaller (by about 8 MB) due to removing Checkstyle from them. This will make using a debug APK in order to help test new features much more friendly to those testing. <sup>[@wb9688 in #3828](https://github.com/TeamNewPipe/NewPipe/pull/3828)</sup>
```

## Conclusion

> 0.19.6 is a pretty nice stability update and also adds some great features. If NewPipe *isn't* stable for you, please consider reporting the issues you are having on [the GitHub repository](https://github.com/TeamNewPipe/NewPipe) (after ensuring the issue isn't already reported), on our IRC channel ([#newpipe on Freenode](https://webchat.freenode.net/?channels=newpipe), or via the built in crash reporter. If you have any other questions feel free to send them in the comments here and someone will reply to you.

Reviewer: *v0.19.6

Reviewer: Also *built-in

## Updating

NewPipe could be built reproducibly for a while now, but starting with this release we also make use of F-Droid's functionality for reproducible builds. F-Droid will thus sign the app with both F-Droid's signature and our signature. This means that new users that will download NewPipe from F-Droid get it signed by our signature and could thus upgrade to the GitHub version, while users that previously downloaded it from F-Droid will still get updates signed by F-Droid.

If you already had NewPipe installed through F-Droid's repository, to get this version of NewPipe you can do one of the following:
- Wait for them to update or

> - Switch to the NewPipe repository by following [these directions](https://newpipe.schabi.org/blog/announcement/f-droid/pinned/f-droid-repo/) (if you had previously installed NewPipe from GitHub releases you will not have to uninstall NewPipe to switch)

Author: I believe the changes in b865d69 resolve this.
